lab-jessica/routes/book-routes.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const storage = require('../lib/storage.js');
+const Book = require('../model/book.js');
+
+module.exports = function(router){
+
+  router.get('/api/books', function(req, res) {
+    res.status(400).send('bad request');
+  });
+
+  router.get('/api/books/:id', function(req, res) {
+    if (req.params.id) {
+      storage.fetchItem('book', req.params.id)
+      .then(book => {
+        res.json(book);
+      })
+      .catch( err => {
+        console.error(err);
+        res.status(404).send('not found');
+      });
+      return;
+    }
+  });
+
+  router.post('/api/books', function(req, res) {
+    try {
+      var book = new Book(req.body.title, req.body.author);
+      storage.createItem('book', book);
+      res.json(book);
+    } catch (err) {
+      console.error(err);
+      res.status(400).send('bad request');
+    }
+  });
+
+  router.post('/api/books', function(req, res) {
```

Comment:
What's this post route for? Looks like you're already accounting for a bad post request with your try catch block above. You shouldn't need it since your standard post request doesn't require any additional query information from the URL.